village/src/components/SmurfForm.js

```diff
@@ -1,54 +1,154 @@
 import React, { Component } from 'react';
+import axios from "axios";
+import { NavLink } from 'react-router-dom';
+import '../App.css';
+
+const smurfData = 'http://localhost:3333/smurfs';
 
 class SmurfForm extends Component {
   constructor(props) {
     super(props);
     this.state = {
       name: '',
       age: '',
-      height: ''
+      height: '',
+      deleteId: '',
+      updateId: '',
+      updateName: '',
+      smurfs: this.props.smurfs
     };
   }
 
+  handleInputChange = e => {
+    this.setState({ [e.target.name]: e.target.value });
+  };
+
   addSmurf = event => {
     event.preventDefault();
-    // add code to create the smurf using the api
-
+    axios.post(smurfData, {
+      name: this.state.name,
+      age: parseInt(this.state.age, 10),
+      height: this.state.height
+    })
+    .then(response => {
+      this.props.setData(response.data);
+    })
+    .catch(error => {
+      console.log(error);
+    });
     this.setState({
       name: '',
       age: '',
-      height: ''
+      height: '',
     });
   }
 
-  handleInputChange = e => {
-    this.setState({ [e.target.name]: e.target.value });
-  };
+  deleteSmurf = (e) => {
+    e.preventDefault();
+    let chooseId = this.state.deleteId;
+    axios.delete(`http://localhost:3333/smurfs/${chooseId}`)
```

Comment:
since you already have a variable declared for localhost link - you can shorten up the URL

```js
`${smurfData}/${chooseId}`
```